src/Model/Item.php

```diff
@@ -1785,19 +1785,14 @@ public static function guidFromUri($uri, $host)
 		// We have to avoid that different routines could accidentally create the same value
 		$parsed = parse_url($uri);
 
-		// We use a hash of the hostname as prefix for the guid
-		$guid_prefix = hash("crc32", $host);
-
 		// Remove the scheme to make sure that "https" and "http" doesn't make a difference
 		unset($parsed["scheme"]);
 
 		// Glue it together to be able to make a hash from it
 		$host_id = implode("/", $parsed);
 
-		// We could use any hash algorithm since it isn't a security issue
-		$host_hash = hash("ripemd128", $host_id);
-
-		return $guid_prefix.$host_hash;
+		// Use a mixture of several hashes to provide some GUID like experience
+		return hash("crc32", $host) . '-'. hash('joaat', $host_id) . '-'. hash('fnv164', $host_id);
```

Comment:
This is cute, but do you know if these fancy hashing algorithms are available on all platforms?

Reply:
We are using them for years to create the GUID for ActivityPub posts.

Reply:
Why don't we use a common guid library to create a guid? I'm not sure if we should create own algorithm for standard use cases because of security/stabilization..

Reply:
The mechanism we are using here ensures that the same arbitrary GUID is crated on all systems for the same post.

Reply:
Indeed, but you switched around the `joaat` and the `fnv164` hashes in this version, is there a specific reason? Can't we have the same GUID generation node-wide?

Reply:
We are using three different ways to generate a GUID:

1. When creating a new post on our system we are using the GUID function in PHP that contains some random noise as well as the system's hostname as CRC32 value.
2. Then we have got the GUID creation for posts from ActivityPub. This is done using the hash values from above but in a different order. This isn't done with random data to provide the same GUID on each Friendica system.
3. And then there is the GUID that we create for posts that are created elsewhere and that we import (like feeds or Twitter posts). We use three different formats to avoid any possible collision (an AP post could have the same URL like a feed from that system and both could exist on the same machine, so they need different GUID formats).

I'm using these hash mechanism because of it's length. So our GUID values are all nearly having the same length.

Reply:
This needs to be documented somewhere, ideally in a centralized method, because it is extremely confusing right now.